### PR TITLE
feat: support for doctype retrieval

### PR DESCRIPTION
**Pull request by EstebanBorai**
Provides support for `DOCTYPE` retrieval in `PullParser`.

### CODE REVIEW

File: src/reader/parser.rs [hunk 1]
```diff
@@ -146,6 +147,12 @@ impl PullParser {
     /// Checks if this parser ignores the end of stream errors.
     pub fn is_ignoring_end_of_stream(&self) -> bool { self.config.c.ignore_end_of_stream }
 
+    /// Retrieves the Doctype from the document if any
+    #[inline]
+    pub fn doctype(&self) -> &Option<String> {
+        &self.data.doctype
```
Comment (kornelski) [inline src/reader/parser.rs:153]: `&Option` should never be used. `Option<&` is equivalent, and more useful in general.

```suggestion
        self.data.doctype.as_deref()
```

File: src/reader/parser/inside_doctype.rs [hunk 2]
```diff
@@ -6,6 +6,10 @@ use super::{DoctypeSubstate, PullParser, QuoteToken, Result, State};
 
 impl PullParser {
     pub fn inside_doctype(&mut self, t: Token, substate: DoctypeSubstate) -> Option<Result> {
+        if let Some(ref mut doctype) = self.data.doctype {
+            doctype.push_str(t.to_string().as_str());
```
Comment (kornelski) [inline src/reader/parser/inside_doctype.rs:10]: Avoids allocation `use fmt::Write`:

```suggestion
            write!(&mut doctype, "{t}").ok()?;
```

**Comment by EstebanBorai** [comment]
@kornelski thanks so much for the suggestions!

**Comment by kornelski** [comment]
The tests are failing

**Review by kornelski — CHANGES_REQUESTED** [review]
Please test your work

**Review by kornelski — CHANGES_REQUESTED** [review]
Please test your work

**Comment by EstebanBorai** [comment]
> Please test your work 

Of course!

**Comment by EstebanBorai** [comment]
Hi @kornelski I think this pr was merged accidentally?

**Pull request merged**